cloudbuild.yaml

```diff
@@ -13,11 +13,185 @@
 # limitations under the License.
 
 
+# This is the Cloud Build build configuration file. It builds and submits the
+# Docker image to Container Registry, and also applies Terraform configurations
+# for the gcloud project based on the current GitHub branch.
+#
+# For dev and prod branches, the following steps are executed:
+# 1. terraform init
+# 2. terraform refresh
+# 3. terraform plan
+# 4. terraform apply
+#
+# For any other branch, the following steps are executed:
+# 1. terraform init for all environments subfolders
+# 2. terraform refresh for all environments subfolders
+# 3. terraform plan for all environment subfolders
+#
+# This is to ensure that changes being proposed hold for every single environment.
+# 
+# To test a build on dev manually:
+# 1. Make sure you are using the correct glcoud project for your dev environment:
+#    gcloud config set project PROJECT_ID
+# 2. Make sure you are on the dev branch:
+#    git checkout dev
+# 3. Build and deploy the project
+#    cd ~/cloud-monitoring-notification-delivery-integration-sample-code
+#    gcloud builds submit --config cloudbuild.yaml .
+#
+# To test a build on prod manually:
+# 1. Make sure you are using the correct glcoud project for your prod environment:
+#    gcloud config set project PROJECT_ID
+# 2. Make sure you are on the prod branch:
+#    git checkout prod
+# 3. Build and deploy the project
+#    cd ~/cloud-monitoring-notification-delivery-integration-sample-code
+#    gcloud builds submit --config cloudbuild.yaml .
+
 steps:
-# Uses the docker build step to build an image called cloud-run-pubsub-service
-- name: 'gcr.io/cloud-builders/docker'
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/cloud-run-pubsub-service', './philips_hue_integration_example']
+- id: 'build docker image'
+  # Uses the docker build step to build an image
+  name: 'gcr.io/cloud-builders/docker'
+  args: ['build', '--tag', 'gcr.io/$PROJECT_ID/${_IMAGE_NAME}', './philips_hue_integration_example']
+  timeout: 300s
```

Comment:
Optional suggestion: It might be nice to future readers to add a comment where seconds is converted to minutes. E.g.

```timout: 300s  # 5 minutes```

Reply:
Done